### PR TITLE
Add Makefile and README usage

### PR DESCRIPTION
**Pull request by ThatcherC**
`usb-can` is a really cool project! Thanks for putting it together! Here are a couple little usability changes I made for personal use that might be of interest - 

- Added a Makefile to make building easy (and give an example gcc command that can be used, even without `make`)
- Added build instructions to README
- Added the help message to README
- Also added a couple example invocations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,3 +11,37 @@ Bus 002 Device 006: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter
...
+  -b BAUDRATE Set TTY/serial BAUDRATE (default: 2000000).
+  -i ID       Inject using ID (specified as hex string).
+  -j DATA     CAN DATA to inject (specified as hex string).
+  -n COUNT    Terminate after COUNT frames (default: infinite).
+  -g MS       Inject sleep gap in MS milliseconds (default: 200 ms).
+  -m MODE     Inject payload MODE (0 = random, 1 = incremental, 2 = fixed).
+```
+
+
+Example commands:
```
Comment (kobolt) [inline README.md]:
```suggestion
## Example commands:
```

File: README.md [hunk 2]
```diff
@@ -11,3 +11,37 @@ Bus 002 Device 006: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter
...
+```
+# dump CAN bus traffic from 1 Mbit CAN bus
+./canusb -t -d /dev/ttyUSB0 -s1000000 -t
```
Comment (kobolt) [inline README.md]: Would like to add a $ and space between the -s and value for consistency.
```suggestion
$ ./canusb -t -d /dev/ttyUSB0 -s 1000000 -t
```

File: README.md [hunk 3]
```diff
@@ -11,3 +11,37 @@ Bus 002 Device 006: ID 1a86:7523 QinHeng Electronics HL-340 USB-Serial adapter
...
+
+# send the bytes 0xBEEE from ID 005 on at 1 Mbit CAN bus
+./canusb -d /dev/ttyUSB0 -s1000000 -t -i 5 -j BEEE
```
Comment (kobolt) [inline README.md]: Same here
```suggestion
$ ./canusb -d /dev/ttyUSB0 -s 1000000 -t -i 5 -j BEEE
```

**Review by kobolt — COMMENTED** [review]
Thanks for the improvements!
I just have some small formatting suggestions for consistency.

**Comment by ThatcherC** [comment]
Thanks for the suggestions! They look good

**Pull request merged**